src/components/slider/daikin-slider.ts

```diff
@@ -0,0 +1,306 @@
+  }
+
+  private _handleDrag = (event: MouseEvent | TouchEvent): void => {
+    const leftDistance = this._calcMousePositionRatio(event);
+    if (leftDistance == null) {
+      return;
+    }
+    const value = getValueFromRatio(this, leftDistance);
+    this.value = value;
+  };
+
+  // This function will triggered when click the slider bar area.
+  private _handleClick(event: MouseEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+    const ratio = this._calcMousePositionRatio(event);
+    if (ratio == null) {
+      return;
+    }
+    const value = getValueFromRatio(this, ratio);
+    this.value = value;
+  }
+
+  // This function is called when an arrow key is pressed while the thumb is focused.
+  private _handleKeyDown(event: KeyboardEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+    const value = getValueByKeydown(this, event.key);
+    if (value == null) {
+      return;
+    }
+    this.value = value;
+  }
+
+  /**
+   * Returns a normalized value of the mouse position on the slider, in the range 0 to 1.
+   *
+   * @param event A MouseEvent.
+   * @returns The mouse's X coordinate normalized to a range of 0 to 1. The left end and beyond of the slider is 0, and the right end and beyond is 1.
+   */
+  private _calcMousePositionRatio(
+    event: MouseEvent | TouchEvent
+  ): number | undefined {
+    const box = this._sliderRef.value?.getBoundingClientRect();
+    if (box == null) {
+      return;
+    }
+    const x =
+      event instanceof MouseEvent
+        ? event.clientX
+        : event.targetTouches[0].clientX;
+    const relativeX = x - box.left;
+    return Math.max(0, Math.min(relativeX / box.width, 1));
+  }
+
+  // This function will triggered when the slider thumb button start dragging.
+  private _startDrag(event: MouseEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+
+    const stopDrag = (): void => {
+      document.removeEventListener("mousemove", this._handleDrag);
+      document.removeEventListener("mouseup", stopDrag);
+    };
+    document.addEventListener("mousemove", this._handleDrag);
+    document.addEventListener("mouseup", stopDrag);
+    this._handleDrag(event);
+  }
+
+  // This function will triggered when the slider thumb button start dragging.
+  private _startTouch(event: TouchEvent) {
+    event.preventDefault();
+    if (this.disabled) {
+      return;
+    }
+
+    const stopTouch = (): void => {
+      document.removeEventListener("touchmove", this._handleDrag);
+      document.removeEventListener("touchend", stopTouch);
+    };
+    document.addEventListener("touchmove", this._handleDrag);
+    document.addEventListener("touchend", stopTouch);
+    this._handleDrag(event);
+  }
+
+  override render() {
+    const progress = this._progress;
+    /* eslint-disable lit-a11y/click-events-have-key-events */
+    return html`
+      <div
```

Comment:
use `<span>` to make slider embeddable to `<span>`.

```suggestion
      <span
```